Userland/Libraries/LibC/stdlib.cpp

```diff
@@ -466,11 +466,10 @@ int unsetenv(char const* name)
 
 int clearenv()
 {
-    size_t environ_size = 0;
-    for (; environ[environ_size]; ++environ_size) {
-        environ[environ_size] = NULL;
+    for (char** env = environ; *env; ++env) {
+        free_environment_variable_if_needed(*env);
+        *env = nullptr;
```

Comment:
This frees the strings pointed to by the env, but it doesn't free the array with all these pointers – see the end of `putenv()` in this function. Should that be freed too?

It's a progression as-is already, so happy to merge this as-is if you prefer that. Let me know.

Thanks for the patch!

Reply:
Thanks for the review.
You can merge it as is, since LibC and LibCore don't expect environ to be null, and the current implementation does not cause memory leaks (unless you overwrite environ yourself.)
You can make another commit to free environ and to replace it with another empty array but I'm not sure if we need to do that.